src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1196 @@ const ActorInit En_Fr_InitVars = {
+        this->actionFunc = EnFr_ChooseJumpFromLogSpot;
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06001534, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06001534), 0, 0.0f);
+    }
+}
+
+void EnFr_ChooseJumpFromLogSpot(EnFr* this, GlobalContext* globalCtx) {
+
+    // Jumping back into water
+    if (sEnFrPointers.flags == 0xC) {
+        this->actor.posRot.rot.y = (32768.0f / M_PI) * sLogSpotToFromWater[this->actor.params].yaw;
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060007BC, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060007BC), 2, 0.0f);
+
+        this->actionFunc = EnFr_JumpingBackIntoWater;
+    }
+
+    // Jumping up
+    else if (this->isJumpingUp) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060007BC, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060007BC), 2, 0.0f);
+        this->actionFunc = EnFr_JumpingUp;
+    }
+}
+
+void EnFr_JumpingUp(EnFr* this, GlobalContext* globalCtx) {
+    f32 yDistToLogSpot;
+
+    if (this->skelAnime.animCurrentFrame == 6.0f) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+
+    } else if (this->skelAnime.animCurrentFrame == 3.0f) {
+        this->actor.gravity = -10.0f;
+        this->actor.velocity.y = 25.0f;
+
+        // Used only for frogs 2
+        if (this->isJumpingToFrogSong) {
+            this->isJumpingToFrogSong = false;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_EAT);
+        }
+    }
+
+    if (EnFr_IsBelowLogSpot(this, &yDistToLogSpot)) {
+        this->isJumpingUp = false;
+        this->actor.gravity = 0.0f;
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060011C0, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060011C0), 0, 0.0f);
+        this->actionFunc = EnFr_ChooseJumpFromLogSpot;
+    } else if ((this->actor.velocity.y <= 0.0f) && (yDistToLogSpot < 40.0f)) {
+        this->skelAnime.animPlaybackSpeed = 1.0f;
+    }
+}
+
+void EnFr_JumpingBackIntoWater(EnFr* this, GlobalContext* globalCtx) {
+    f32 y_underwater;
+
+    y_underwater = sLogSpotToFromWater[this->actor.params].yDist + this->posLogSpot.y;
+
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.posRot.rot.y, 2, 10000, 100);
+
+    if (this->skelAnime.animCurrentFrame == 6.0f) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+    } else if (this->skelAnime.animCurrentFrame == 3.0f) {
+        this->actor.speedXZ = 6.0f;
+        this->actor.gravity = -10.0f;
+        this->actor.velocity.y = 25.0f;
+    }
+
+    // Final Spot Reached
+    if ((this->actor.velocity.y < 0.0f) && (this->actor.posRot.pos.y < y_underwater)) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06001534, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06001534), 0, 0.0f);
+        this->actionFunc = EnFr_InitJumpingOutOfWater;
+        EnFr_DrawIdle(this);
+        this->isDeactivating = true;
+        EnFr_OrientUnderwater(this);
+    }
+}
+
+void EnFr_SetScaleActive(EnFr* this, GlobalContext* globalCtx) {
+    if (this->isGrowing) {
+        if (this->isGrowing != true) {
```

Comment:
No, it's only ever set as 0 or 1 but I agree that it's suspicious. I was able to put it into a switch case form that looks a lot cleaner than before, with the two cases being true and false. Would it be better to bring it back to 0/1 since it's in a switch case? Or is true/false still fine